src/virtual.jl

```diff
@@ -311,6 +311,56 @@ function _get_integrated_master(b)
   return check
 end
 
+function get_cavity_frequency(ele::LineElement, key::Symbol)
+  c = ele.RFParams
+  if isnothing(c)
+    error("Unable to get $key of LineElement $(ele.name): No RFParams present")
+  end
+  return @noinline _get_cavity_frequency(c, key)
+end
+
+function _get_cavity_frequency(c, key)
+  harmon_master = CAVITY_FREQUENCY_MAP[key]
+  if c.harmon_master == harmon_master
+    return c.frequency
+  else
+    correctkey = CAVITY_FREQUENCY_INVERSE_MAP[c.harmon_master]
+    error("Cannot calculate $key of RFParams since particle species is unknown at Beamlines level and harmon_master=$(c.harmon_master)")
+  end
+end
+
+function set_cavity_frequency!(ele::LineElement, key::Symbol, value)
+  if !haskey(ele.pdict, RFParams)
+    harmon_master = CAVITY_FREQUENCY_MAP[key]
+    setindex!(ele.pdict, RFParams(harmon_master=harmon_master), RFParams)
+  end
```

Comment:
```suggestion
r = ele.RFParams
if isnothing(r)
  ele.RFParams = RFParams(harmon_master=key == :harmon ? true : false)
end
```

Accessing a `pdict` directly is no longer allowed, this is the new proper way to add a parameter group